frontend/test/assignmentModal.test.tsx

```diff
@@ -0,0 +1,116 @@
+import "@testing-library/jest-dom";
+import mockAxios from "jest-mock-axios";
+import {
+    getBaseOkResponse,
+    getBaseProject,
+    getBaseProjectSkill,
+    getBaseSkillType,
+    getBaseStudent,
+    getBaseUser,
+} from "./TestEntityProvider";
+import { act, render, screen, waitFor } from "@testing-library/react";
+import { makeCacheFree } from "./Provide";
+import AssignmentModal from "../src/components/project_assignment/assignmentModal";
+import { Dispatch } from "react";
+import userEvent from "@testing-library/user-event";
+import { AxiosResponse } from "axios";
+import { UserRole } from "../src/api/entities/UserEntity";
+import { Assignment } from "../src/api/entities/AssignmentEntity";
+
+jest.mock("next/router", () => require("next-router-mock"));
+
+afterEach(() => {
+    mockAxios.reset();
+});
+
+type SkillInfo = { skillName: string; skillColor: string; skillUrl: string };
+type ModalInfo = { studentName: string; studentUrl: string; skillInfo: SkillInfo; projectName: string };
```

Comment:
Is this type only needed in tests? Don't we need it anywhere else then ass well? (no idea here)